admin/pre-reg/views.py

```diff
@@ -0,0 +1,171 @@
+    for reviewer in users:
+        if (is_in_prereg_group(reviewer)):
+            reviewers.append(str(reviewer.username))
+    return reviewers
+
+def is_in_prereg_group(user):
+    """Determines whether a user is in the prereg_group
+    :param user: User wanting access to prereg material
+    :return: True if prereg False if not
+    """
+    return user.groups.filter(name='prereg_group').exists()
+
+@login_required
+# @user_passes_test(is_in_prereg_group)
+def prereg(request):
+    """Redirects to prereg page if user has prereg access
+    :param request: Current logged in user
+    :return: Redirect to prereg page with username, reviewers, and user obj
+    """
+    #prereg_admin = request.user.has_perm('auth.prereg_admin')
+    #user = {
+    #     'username': str(request.user.username),
+    #     'admin': json.dumps(prereg_admin)
+    # }
+    user = {
+        'username': 'user_placeholder',
+        'admin': 'admin_placeholder'
+    }
+    #reviewers = get_prereg_users()
+    reviewers = ['admin_placeholder']
+
+    #context = {'user_info': user, 'reviewers': reviewers, 'user': request.user}
+    context = {
+        'user_info': user,
+        'reviewers': reviewers
+    }
+    return render(request, 'pre-reg/prereg.html', context)
+
+@login_required
+# @user_passes_test(is_in_prereg_group)
+def prereg_form(request, draft_pk):
+    """Redirects to prereg form review page if user has prereg access
+    :param draft_pk: Unique id for selected draft
+    :return: Redirect to prereg form review page with draft obj
+    """
+    draft = get_draft_or_error(draft_pk)
+    context = {
+        'draft': utils.serialize_draft_registration(draft)
+    }
+    return render(request, 'pre-reg/edit_draft_registration.html', context)
+
+@login_required
+# @user_passes_test(is_in_prereg_group)
+def get_drafts(request):
+    """Determines whether a user is in the general_administrator_group
+    :param user: User wanting access to administrator material
+    :return: True if general administrator False if not
+    """
+    prereg_schema = MetaSchema.find_one(
+        Q('name', 'eq', 'Prereg Challenge') &
+        Q('schema_version', 'eq', 2)
+    )
+    all_drafts = DraftRegistration.find(
+        Q('registration_schema', 'eq', prereg_schema) &
+        Q('approval', 'ne', None)
+    )
+    serialized_drafts = {
+        'drafts': [utils.serialize_draft_registration(d) for d in all_drafts]
+    }
+    return JsonResponse(
+        serialized_drafts
+    )
+
+@login_required
+# @user_passes_test(is_in_prereg_group)
+@csrf_exempt
+def approve_draft(request, draft_pk):
+    """Approves current draft
+    :param user: Current logged in user
+    :param draft_pk: Unique id for current draft
+    :return: DraftRegistrationApproval obj
+    """
+    draft = get_draft_or_error(draft_pk)
+
+    user = load_osf_user(request.user)
+    try:
+        draft.approve(user)
+    except PermissionsError:
+        draft.approval.add_authorizer(user, save=True)
```

Comment:
@cwisecarver if https://github.com/samchrisinger/osf.io/commit/11146cc8806750841ae4bcda3a146b3b08429a20 goes beyond the scope of this PR I'll pull it out.